File: export_from_firebase.py
import firebase_admin
from firebase_admin import credentials, firestore, storage
import json
import os


# Initialize Firebase
cred = credentials.Certificate("<your-service-account-key>.json")
firebase_admin.initialize_app(cred, {
    'storageBucket': '<your-buket-name>'
})


db = firestore.client()
bucket = storage.bucket()


# Export Firestore Collections
collections = db.collections()
for collection in collections:
    docs = collection.stream()
    data = [{**doc.to_dict(), "id": doc.id} for doc in docs]
    os.makedirs(f'collections', exist_ok=True)
    with open(f'collections/{collection.id}.json', 'w') as file:
        json.dump(data, file, indent=2)


# Export Files from Firebase Storage
blobs = bucket.list_blobs()
for blob in blobs:
    directory = os.path.dirname(f'bucket/{blob.name}')
    if not os.path.exists(directory):
        os.makedirs(directory)
    blob.download_to_filename(f'bucket/{blob.name}')
    print(f'Downloaded {blob.name} to bucket/{blob.name}')

File: node-scripts-equivalent/export_from_firebase.js
var admin = require("firebase-admin");
var fs = require('fs');
const path = require('path');
var serviceAccount = require("./<your-service-account-key>.json");

admin.initializeApp({
    credential: admin.credential.cert(serviceAccount),
});


const db = admin.firestore();
const bucket = admin.storage().bucket("gs://<your-buket-name>");


db.listCollections().then(collections => {
    collections.forEach(collection => {
        collection.get().then(docs => {
            const data = docs.docs.map(doc => ({ id: doc.id, ...doc.data() }));
            fs.writeFileSync(`collections/${collection.id}.json`, JSON.stringify(data, null, 2));
        });
    });
});


bucket.getFiles().then((files) => {
    if (files[0].length === 0) {
        console.log('No files to download.');
        return;
    }

    files[0].forEach((file) => {
        const destinationPath = `bucket/${file.name}`;
        const directory = path.dirname(destinationPath);

        if (!fs.existsSync(directory)) {
            fs.mkdirSync(directory, { recursive: true });
        }

        file.download({ destination: destinationPath })
            .then(() => console.log(`Downloaded ${file.name} to ${destinationPath}`))
            .catch((error) => console.error(`Error downloading ${file.name}:`, error));
    });
}).catch((error) => console.error('Error getting files:', error));
